# Loading Libraries

In [2]:
options(warn=-1)
library(dplyr)
library(mltools)
library(data.table)
library(ggplot2)
library(flextable)

# Importing Dataset

The data set is imported using read.csv() function. By using the str() function we can
print the all the variables of the data set.

In [2]:
superStore <- read.csv("Superstore.csv")

str(superStore)

'data.frame':	9994 obs. of  20 variables:
 $ Order.ID     : chr  "CA-2016-152156" "CA-2016-152156" "CA-2016-138688" "US-2015-108966" ...
 $ Order.Date   : chr  "11/08/2016" "11/08/2016" "06/12/2016" "10/11/2015" ...
 $ Ship.Date    : chr  "11/11/2016" "11/11/2016" "6/16/2016" "10/18/2015" ...
 $ Ship.Mode    : chr  "Second Class" "Second Class" "second class" "Standard Class" ...
 $ Customer.ID  : chr  "CG-12520" "CG-12520" "DV-13045" "SO-20335" ...
 $ Customer.Name: chr  "Claire Gute" "Claire Gute" "Darrin Van Huff" "Sean O'Donnell" ...
 $ Segment      : chr  "Consumer" "Consumer" "Corporate" "Consumer" ...
 $ Country      : chr  "United States" "United States" "United States" "United States" ...
 $ City         : chr  "Henderson" "Henderson" "Los Angeles" "Fort Lauderdale" ...
 $ State        : chr  "Kentucky" "Kentucky" "California" "Florida" ...
 $ Postal.Code  : int  42420 42420 90036 33311 33311 90032 90032 90032 90032 90032 ...
 $ Region       : chr  "South" "South" "West" "Sout

# 1 Introduction

The objective of the task is to apply the six information quality aspects to one of the picked
datasets to investigate and upgrade its quality. The 6 information quality aspects were
applied to the super store data set to assess its quality and, subsequently, proven methods for information quality
aspects were utilized at whatever point important to sanitize the information and accomplish a more serious level of 
value. Fulfillment, Conformity,Consistency, Accuracy, Duplicates, and Integrity are the six information quality
aspects that would be suggested in this paper.

###  Dataset

This dataset is very self explanatory.It contains data about the product sales of a global superstore.
The data set have 9994 observations (rows) with 20 variables


### Data Dictionary

*Order id:Contains the order's ID
*Order Date:Contains the order date 
*Ship Date:Contains the shipping date
*Ship mode:Contains the class of the transport
*Customer id:Contains the customer id
*Customer name:Stores the customer name
*Segment:Contains the segment of the product
*Country:Contains the country of the order
*City:Contains the city of the order
*State:Contains the state of the order
*Postal Code:Contains the postal code of the order's destination
*Region:Contains the region for the order
*Product Id:Contains the product ID
*Category:Contains the product category
*Sub category:Contains the product sub category
*Product Name:Contains the product name
*Sales:Contains the sales of a product
*Quantity:Contains the quantity of a product
*Discount:Contains the discount of a product
*Profit:Contains the profit of a product

# 2 Completeness

### 2.1 Counting missing values

The blank cells or NAs were found in two factors: postal code and amount.
Postal code and amount factors have 9, 3 missing values, separately

In [4]:
df_withna <- superStore %>% filter_all(any_vars(is.na(.)))
summary(df_withna)

   Order.ID          Order.Date         Ship.Date          Ship.Mode        
 Length:12          Length:12          Length:12          Length:12         
 Class :character   Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character   Mode  :character  
                                                                            
                                                                            
                                                                            
                                                                            
 Customer.ID        Customer.Name        Segment            Country         
 Length:12          Length:12          Length:12          Length:12         
 Class :character   Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character   Mode  :character  
                                                                            

### 2.2 Replacing missing values 

For variable postal code and amount factors the missing values are replaced  with zero
to demonstrate no postal code and no amount for those particular variables.

In [5]:
superStore$Postal.Code[is.na(superStore$Postal.Code)] <- 0
superStore$Quantity[is.na(superStore$Quantity)] <- 0

### 2.3 Correcting variable labels

There was one more issue in  ship mode, segment and region labels. For the ship mode section and  
segment column had a portion of the labels utilized were lower case and others capitalized .
The region segment had labels utilized for the west classification were W, the south classification 
were S, the central class C and the east classification were E. Along these columns, there is need
to replace this with the right ones.

In [20]:
flextable(filter(superStore, Region == "C" | Region == "E" | Region == "S" | Region == "W"))

a flextable object.
col_keys: `Order.ID`, `Order.Date`, `Ship.Date`, `Ship.Mode`, `Customer.ID`, `Customer.Name`, `Segment`, `Country`, `City`, `State`, `Postal.Code`, `Region`, `Product.ID`, `Category`, `Sub.Category`, `Product.Name`, `Sales`, `Quantity`, `Discount`, `Profit` 
header has 1 row(s) 
body has 11 row(s) 
original dataset sample: 
        Order.ID Order.Date  Ship.Date      Ship.Mode Customer.ID
1 CA-2014-115812 06/09/2014  6/14/2014 Standard Class    BH-11710
2 CA-2016-137330 12/09/2016 12/13/2016 Standard Class    KB-16585
3 CA-2016-118255 03/11/2016  3/13/2016    First Class    ON-18715
4 CA-2016-111682  6/17/2016  6/18/2016    First Class    TB-21055
5 US-2015-134026  4/26/2015 05/02/2015 Standard Class    JE-15745
    Customer.Name   Segment       Country        City      State Postal.Code
1 Brosina Hoffman  Consumer United States Los Angeles California       90032
2       Ken Black corporate United States     Fremont   Nebraska       68025
3   Odella Nelson Corporate

In [9]:
superStore$Ship.Mode[superStore$Ship.Mode == "first class"] <- "First Class"
superStore$S[superStore$Ship.Mode == "second class"] <-  "Second Class"
superStore$Ship.Mode[superStore$Ship.Mode == "standard class"] <- "Standard Class"
superStore$Ship.Mode[superStore$Ship.Mode == "Standard class"] <- "Standard Class"
superStore$Ship.Mode[superStore$Ship.Mode == "standard Class"] <- "Standard Class"

superStore$Segment[superStore$Segment == "consumer"] <- "Consumer"
superStore$Segment[superStore$Segment == "corporate"] <- "Corporate"
superStore$Segment[superStore$Segment == "home office"] <- "Home Office"

superStore$Region[superStore$Region == "C"] <- "Central"
superStore$Region[superStore$Region == "E"] <- "East"
superStore$Region[superStore$Region == "S"] <- "South"
superStore$Region[superStore$Region == "W"] <- "West"

superStore$State[superStore$State == "FL"] <- "Florida"
superStore$State[superStore$State == "CL"] <- "Colorado"

superStore$Country[superStore$Country == "US"] <- "United States"

### 2.4 Setting Decimal Places

The sales columnn and the quantity column have two decimal places but in real sense the 
sales and quantity columns have to be whole numbers to incidate an exact amount.

In [3]:
round(superStore$Sales)
round(superStore$Quantity)

[1]   262   732    15   958    22    49     7   907    19   115  1706   911
  [13]    16   408    69     3   666    56     9   213    23    19    60    71
  [25]  1045    12    91  3083    10   124     3    86     7    16    29  1098
  [37]   191   113   532   212   371   147    78    96    46    17   212    45
  [49]    22    38    75     6    90    15  1030   209    32   319    15    30
  [61]    48     2    14    26   147    80   213  1113   168    76     5    19
  [73]   832    97    73     1    10    27    19   208    17    15    21   201
  [85]   230   302    20     6   158    20    74     6    13    53    33     6
  [97]    97    51    78    65    96     2    24   239   102    37    74    28
 [109]     3   340    42    76    27    40     5    24   130   788   158    47
 [121]    31   227   115    68   601   618     2   244    81   239    60    78
 [133]    21    20    35    12     4    76    66    43    83     9    11   144
 [145]   839   672    94   384   150  1952   172   158   203    58   106    81
 [157]     7   458    15   945     6    54    28    28    10  8160   276  1740
 [169]    32   178   144    21   111   340    52    20    97   397    16     3
 [181]    25   409   504   150    29     7   177    37    20   899    72    52
 [193]   626    20    14     7     6    46     3    16    22   219     3    66
 [205]    35   445    84   132    16    52    92    21    24   452    63  1188
 [217]    90    93   302     6    23    20    73   480    27     2   622    22
 [229]   162   390    19   234   621     5   258   618    11    26   420    12
 [241]    32   177     4     7  2002   167    48  1503    26   322     8  3347
 [253]    81   362    12    82    54   648    20   222    18     2  3060  2520
 [265]   328    80    14     8    37    58   726   210     5    11     9     6
 [277]    12    18    60    25     2  1114  1039   142   240    31   254   195
 [289]   961    19    18   256    87   300   230   218    79    28    32  1082
 [301]    57    78    14   219    27    10    45   289     5    15   467    15
 [313]     6    88   178    16    99   136  3992   276   360    44     7   252
 [325]   100    16   291    54   787   100    38    83    21     5     5   960
 [337]    14     8   698     5    18   510    31    72    89    48     8    25
 [349]    13     7    25     3    49  4355   389     8    17    34    36   648
 [361]    21    21   489     6    47   212    23     7   105  1044    26    53
 [373]     8  1024     9   479    99  1488     9    24    12    51    50    42
 [385]   375    84   482     3     3    23    40   246  1800    12    76    50
 [397]    13    70    36  2396   131    58    10    39    36   180  1200    27
 [409]  1004    10    28    56  1337   114   140   307    96   384     6     9
 [421]    15   197    57    33   866    28   288    70     7   190   409   292
 [433]     5   714     5   248  1008   313    32   208    12   195    71    91
 [445]   243    22     3    60   162   264    30    10    35   284   665    64
 [457]   130   748     9   104   900    51    24  1273    28   185    78   255
 [469]   205     5    55   341   223   704    93    63    95    96    40    15
 [481]   704     9     6   160    30   514   280  2736     8    64    70   449
 [493]    11    94   190   105   120   256   242    69    23    15   802     2
 [505]    33    16     3     8    47  6355   126    38     7     7     6  3000
 [517]    51    12  1126    18   130    14   211    55    66    22   684    46
 [529]    36   666    53   191    48   980    23    17    43    11   153   284
 [541]   469   381   647    58   101    66    41    13   250    11     9  1122
 [553]    35    11  1296    19    21  1336    32    43    84    13    13     4
 [565]     3   374    92    81    19   451    72    14    33    15     9   161
 [577]    20     7    70     9    52   470   106    31     7   406    71   295
 [589]    85    21    17    10     9    31    76  1200   446   328    12   144
 [601]   494     6   143    46     7    43   132     

[1]      2      3      2      5      2      7      4      6      3      5
  [11]      9      4      3      3      5      3      6      2      2      3
  [21]      4      7      7      2      3      2      3      7      2      3
  [31]      2      6      6  10000      3      7      5      9      3      3
  [41]      4      4      2      2      2      2      4      3      2      6
  [51]      6      2      1      7      5      6      5      5      2      2
  [61]      4      1      2      6      3      4      5      8      4      2
  [71]      1      3      8      2      1      3      3      3      5      1
  [81]      3      5      1      7      3      2      1      2      7      3
  [91]      2      1      2      3      2   1000      7      3      6      7
 [101]      3      3      4      6      3      3      8      1      1      5
 [111]      2      2      6     14      2      3      6      3      1      3
 [121]      4      6      9      7      3      6      2      7      2      3
 [131]      5      2      9      3      1      4      2      3      6     14
 [141]      2      3      3      3      3      7      4     11      3      8
 [151]      5      5      2      7      4      6      3      2      2      7
 [161]      1      2      5      2      3      8      3      9      6      5
 [171] 500000      3      2      7      2      4      4      7      5      1
 [181]      2      7      4      5      2      2      8      3      3      4
 [191]      6      8      3      5      7      2      3      3      2      3
 [201]      3      2      1      2      7      4      4      2      4      9
 [211]      1      2      2      5      3      9      2      6      3      2
 [221]      6      4      4      4      2      1      5      1      2      8
 [231]      7      2      3      2      3      3      2      5      5      3
 [241]   2347      5      3      2      7      2      6      5      4      2
 [251]      1     13      6      4      9      2      5      6      3      3
 [261]      5      2      3      7     NA      2      3      6      1      3
 [271]      4      7      3      3      2      1      2      4      6      3
 [281]      5      7      5      5      5      6      7      2      5      7
 [291]      8      2      3      8      3      3      5      3      5      8
 [301]      3      4      1      3      2      3      3      6      1      4
 [311]      2      1      3      3      2      3      4      1      2      6
 [321]      4     NA      2      6      1      1      3      2      4     10
 [331]      6      2      2      3      1      2      4      2      3      4
 [341]      4     10     13     12      4      3      3      2      2      1
 [351]      2      1      4      4      6      2      4      5      5      8
 [361]      2      2      3      2      8      4      4      2      7      4
 [371]      5      3      5      8      1     10      4      7      3      3
 [381]      4      7      3      2      3      3      4      1      1      4
 [391]      2      2      3      3      3      2      2      4      3     NA
 [401]      4      2      2      6      3      5      3      5      7      1
 [411]      9      1     13      2      7      4      8      5      1      4
 [421]      1      6      4      3      4      2      8      1      6      2
 [431]      7      5      2      5      2      5      3      7      8      3
 [441]      1      3      3      4      3      7      1      3      3      6
 [451]      3      3      4      2      2      4      2      3      2      4
 [461]      9      3      5      6      5      2      2      7      2      3
 [471]      1      3      2      4      6      7      5      6      3      5
 [481]      5      3      2      2      2      5      5      6      1      2
 [491]      2      5      3      2      3      2      8      4      2      9
 [501]      2      2      2      3      1      5      1      3      3      5
 [511]      3      2      3      3      2      5      3      2      3      1

# 3 Conformity

By looking at the structure of the variables we can see that the factors having right design. 
The data type for every factor is underneath where postal code, sales, quantity, discount and profit
are numeric while the other factors are character factors.

In [4]:
str(superStore)

'data.frame':	9994 obs. of  20 variables:
 $ Order.ID     : chr  "CA-2016-152156" "CA-2016-152156" "CA-2016-138688" "US-2015-108966" ...
 $ Order.Date   : chr  "11/08/2016" "11/08/2016" "06/12/2016" "10/11/2015" ...
 $ Ship.Date    : chr  "11/11/2016" "11/11/2016" "6/16/2016" "10/18/2015" ...
 $ Ship.Mode    : chr  "Second Class" "Second Class" "second class" "Standard Class" ...
 $ Customer.ID  : chr  "CG-12520" "CG-12520" "DV-13045" "SO-20335" ...
 $ Customer.Name: chr  "Claire Gute" "Claire Gute" "Darrin Van Huff" "Sean O'Donnell" ...
 $ Segment      : chr  "Consumer" "Consumer" "Corporate" "Consumer" ...
 $ Country      : chr  "United States" "United States" "United States" "United States" ...
 $ City         : chr  "Henderson" "Henderson" "Los Angeles" "Fort Lauderdale" ...
 $ State        : chr  "Kentucky" "Kentucky" "California" "Florida" ...
 $ Postal.Code  : int  42420 42420 90036 33311 33311 90032 90032 90032 90032 90032 ...
 $ Region       : chr  "South" "South" "West" "Sout

# 4 Consistency

The values for profits have a major issue in consistency. The profits are not steady under one.

In [22]:
flextable(filter(superStore, Profit < 0))

a flextable object.
col_keys: `Order.ID`, `Order.Date`, `Ship.Date`, `Ship.Mode`, `Customer.ID`, `Customer.Name`, `Segment`, `Country`, `City`, `State`, `Postal.Code`, `Region`, `Product.ID`, `Category`, `Sub.Category`, `Product.Name`, `Sales`, `Quantity`, `Discount`, `Profit` 
header has 1 row(s) 
body has 1871 row(s) 
original dataset sample: 
        Order.ID Order.Date  Ship.Date      Ship.Mode Customer.ID
1 US-2015-108966 10/11/2015 10/18/2015 Standard Class    SO-20335
2 US-2015-118983 11/22/2015 11/26/2015 Standard Class    HP-14815
3 US-2015-118983 11/22/2015 11/26/2015 Standard Class    HP-14815
4 US-2017-156909  7/16/2017  7/18/2017   Second Class    SF-20065
5 US-2015-150630  9/17/2015  9/21/2015 Standard Class    TB-21520
    Customer.Name     Segment       Country            City        State
1  Sean O'Donnell    Consumer United States Fort Lauderdale      Florida
2   Harold Pawlan Home Office United States      Fort Worth        Texas
3   Harold Pawlan Home Office United 

We can create a subset of the data by removing these inconsistent number of profits.

In [24]:
superStore <- subset(superStore, Profit >= 0)


# 5 Accuracy

The data in this dataset is accurate as it matches the various categories used.

In [29]:
flextable(head(filter(superStore, Region == "South")))

a flextable object.
col_keys: `Order.ID`, `Order.Date`, `Ship.Date`, `Ship.Mode`, `Customer.ID`, `Customer.Name`, `Segment`, `Country`, `City`, `State`, `Postal.Code`, `Region`, `Product.ID`, `Category`, `Sub.Category`, `Product.Name`, `Sales`, `Quantity`, `Discount`, `Profit` 
header has 1 row(s) 
body has 6 row(s) 
original dataset sample: 
        Order.ID Order.Date  Ship.Date      Ship.Mode Customer.ID
1 US-2017-100930 04/07/2017 04/12/2017 Standard Class    CS-12400
2 CA-2014-136567 12/20/2014 12/21/2014    First Class    PS-19045
3 CA-2016-144344 10/28/2016 10/28/2016       Same Day    PG-18820
4 CA-2014-104283  6/27/2014 07/01/2014 Standard Class    LM-17065
5 US-2014-155502  1/26/2014  1/31/2014 Standard Class    SD-20485
       Customer.Name     Segment       Country          City       State
1 Christopher Schild Home Office United States         Tampa     Florida
2    Penelope Sewall Home Office United States  Harrisonburg    Virginia
3    Patrick Gardner    Consumer United 

# 6 Duplicates

There are no duplicate entries in the data set.

In [30]:
superStore %>%
transmute(duplicate = duplicated(.)) %>%
filter(duplicate == "TRUE")

duplicate

# 7 Intergrity

The data has no intergral issues.When the integrity of data is secure, the information stored will
remain complete, accurate, and reliable no matter how long it’s stored or how often it’s accessed.

# 8 Conclusion

It is exceptionally vital to clean the data set before examination and perceptions. The super store data set 
was cleaned by utilizing six layered information quality ascribes. The data set was found have missing qualities
and was cleaned. A few issues were found as to ship mode, segment and region variable having two qualities for
a similar classification and were rectified.
The information was conformant and put away in appropriate configuration. There was a serious issue in 
profits which were conflicting under 0, and were eliminated. No copied sections were found in the dataset.
There is need to appropriately investigate the informational index and may address the class or may erase these columns
to abstain from any deceptive data during information examination and representations.